active/Resources/StyleVersion2.md

```diff
@@ -0,0 +1,89 @@
+ThemeVersion API
+=
+
+# Background
+
+XAML (WinUI) has a resources feature where you can put objects in a "resource dictionary", then
+use the resources by referencing the key name. All of the styling for XAML controls
+is defined in a resource dictionary and referenced by the controls.
+But an application can override these by defining a resource dictionary with one or more matching keys.
+
+XAML ships such an overrides dictionary named
+[XamlControlsResources](http://msdn.microsoft.com/library/Microsoft.UI.Xaml.Controls.XamlControlsResources).
+This dictionary defines all the resources necessary for all of the built-in controls.
+Since the resources in this dictionary can change over time, a new `ThemeVersion` property is being added.

```

Comment:
The word "theme" seems to be used for a bunch of related (but different) concepts.

1. [XamlUIPresenter.ThemeKey](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Hosting.XamlUIPresenter.ThemeKey?view=winrt-19041) / [XamlUIPresenter.ThemeResourcesXaml](https://docs.microsoft.com/uwp/api/windows.ui.xaml.hosting.xamluipresenter.themeresourcesxaml?view=winrt-19041) / [ResourceDictionary.ThemeDictionaries](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.ResourceDictionary.ThemeDictionaries?view=winrt-19041)

This sense of "theme" appears to be "choosing a portion of the combined resource dictionary to apply to the app."

2. [Application.RequestedTheme](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Application.RequestedTheme?view=winrt-19041)

This sense of "theme" appears to be "choosing between light or dark default styling."

3. XamlControlsResources.ThemeVersion (introduced here)

This sense of theme appears to be "choosing a historical styling set."

The new sense is similar to number 2, in that they both mean "Completely swapping out a set of resources."

So you could pick a ThemeVersion = Version2, and a RequestedTheme = Light, and you get "Version 2 with light parts."

Maybe this is really a "StyleSet"?

Reply:
How does this differ from MRT?  Getting different resources for different scales or languages?

Reply:
Unfortunately Xaml and the [ResourceManager](https://docs.microsoft.com/uwp/api/Windows.ApplicationModel.Resources.Core.ResourceManager) have independent features both named "resources".  Adapting to resolution scales and languages are generally done in RESW, Xaml resources are used for sharing objects such as sharing a Brush between controls.

Regarding "StyleSet", the original name we had for this was "StylesVersion", but my confusion with that name is that the resource dictionary has more than Style objects in it, it has a lot of brushes and colors and numerics as well.

But in addition to that list of "theme" terminology is "{ThemeResource}".

Brainstorming:
ThemeVersion
StylesVersion
StyleSet
ResourcesVersion
ControlsResourcesVersion
?

Reply:
DefaultResourcesVersion? ("What version of the default resources do you want?" "Oh, I want the version that came with WinUI 2.1.")

Reply:
I know _why_ we're having this conversation, but the notion of a "version" is that it's nominally linear and you can compare them to know when X is better (greater) than Y.  Maybe a compromise is `ThemeStyle` or `StyleTheme` ? Something that we won't regret when we want to say that "apples are not better than oranges, they're simply different."

Reply:
In this case they are linear, but I don't think there are any comparison scenarios. You typically don't set it, but maybe you have a custom set of resources that's designed to align with Version3, so you want to set this property to Version3.

By ThemeStyle, you mean e.g. ThemeStyle.Version2?

Reply:
Recommend:
Since this is the version of the XamlControlsResources, name this enum ControlsResourcesVersion.